hnn_core/network.py

```diff
@@ -140,33 +142,36 @@ class Network(object):
     ----------
     params : dict
         The parameters to use for constructing the network.
+    initialise_hnn_drives : bool
+        If True (default), attach and instantiate drives as in HNN-GUI
```

Comment:
I think this should be a separate function, not mangled as part of the `Network object`. In that function, you instantiate the `Network` object and add the default drives through whatever API you design

Reply:
Basically, don't mix instantiation with class definition

Reply:
Set as a parameter to allow dozens of existing tests to pass without modification. Separate PR?